Refactor 'PRODUCTION' boolean environment variable to a key-value pair

gdiazc opened the pull request:
'APP_ENV' should be set to 'PRODUCTION' in production.

gdiazc committed:
Refactor 'PRODUCTION' boolean environment variable to a key-value pair
'APP_ENV' set to 'PRODUCTION'.

gdiazc committed:
Be sure to retrieve correctly formatted tweets from Elasticsearch.

gdiazc committed:
Use pandas to add a second timeseries plot to the UI.

gdiazc committed:
UI tweaks.

gdiazc committed:
Add nltk and corpus.

dev1-fund3 commented on app.py:
```diff
@@ -38,8 +39,14 @@ def get_tweets():
-    xs = []
-    ys = []
+
+    import dateutil
```
move imports to top of file.

gdiazc replied:
Resolved in next commit!

dev1-fund3 commented on app.py:
```diff
@@ -38,8 +39,14 @@ def get_tweets():
+    id_strs = []
+    created_ats = []
+    full_texts = []
+    subjs = []
```
use more descriptive variable names. Is this subjects? and below polarities?

gdiazc replied:
Resolved in next commit!

dev1-fund3 commented on app.py:
```diff
@@ -51,17 +58,38 @@ def get_tweets():
+        'polarity': polas
+    })
 
     fig = bokeh.plotting.figure(plot_width=600, plot_height=400)
```
create a separate method for plotting the dataframe

gdiazc replied:
Resolved in next commit!

gdiazc committed:
Resolve review requests:

app.py:
- imports to top of file
- refactor var names
- move plotting logic to separate file

gdiazc committed:
Read P_CRON_NTWEETS environment variable for cron job.

dev1-fund3 commented on bin/cron_pull_tweets.py:
```diff
@@ -67,4 +67,8 @@ def run_task(params, ntweets=3, maxpages=2):
             params = json.load(f)
 
-    run_task(params)
+    ntweets = 100
```
Assuming this is used for debugging? If so, make an env var

gdiazc replied:
The env var 'P_CRON_NTWEETS' was created (see lines 71–74). Line 70 serves as a fallback default, although perhaps we could refactor that!

dev1-fund3 reviewed (commented):
former concerns addressed. 1 question for clarification

calvinleungyk commented on app.py:
```diff
@@ -1,24 +1,24 @@
-def get_tweets():
-    if 'PRODUCTION' in os.environ:
+def _get_env_params():
+    if 'APP_ENV' in os.environ and os.environ['APP_ENV'] == 'PRODUCTION':
```
minor nitpick: I think you can provide a default value with `os.getenv('APP_ENV', 'default_value_of_something')` and so you don't need that `if 'APP_ENV' in os.environ` check

gdiazc replied:
Ahh yes, I should have known! I will most definitely use that in the next commit.

gdiazc replied:
Ready!

calvinleungyk commented on bin/cron_pull_tweets.py:
```diff
@@ -67,4 +67,8 @@ def run_task(params, ntweets=3, maxpages=2):
 
-    run_task(params)
+    ntweets = 100
+    if 'P_CRON_NTWEETS' in os.environ:
```
Same comment for `os.getenv()`

gdiazc replied:
Incorporated! See [
f68b21a](https://github.com/fund3/sentiment-dashboard/pull/3/commits/f68b21a7325889dba9d66e7322195487eb687ab7)

calvinleungyk commented on app.py:
```diff
@@ -1,24 +1,24 @@
 from flask import Flask, render_template
 import os
 import simplejson as json
```
Sort and format the imports as follows: http://google.github.io/styleguide/pyguide.html#313-imports-formatting

gdiazc replied:
Ready! See [564fbce](https://github.com/fund3/sentiment-dashboard/pull/3/commits/564fbce1529837aff9d8f5d1000d3bb72073b28b)

calvinleungyk reviewed (changes requested):
Seems like you should add a yaml config file and read from yaml instead of checking env?  Looks like you have a couple instances of doing that and I anticipate more flags or envvars that you have to set

gdiazc committed:
Sort tweets by created_at and limit number of tweets.

gdiazc committed:
Streamline UI.

gdiazc committed:
Get APP_ENV env var with a default

gdiazc committed:
Get APP_ENV env var with a default and clean code for P_CRON_NTWEETS
param.

gdiazc committed:
Sort and update imports and requirements
